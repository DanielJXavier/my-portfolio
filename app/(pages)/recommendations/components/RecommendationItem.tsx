import Image from "next/image";

import { RecommendationsType } from "../data";

export default function RecommendationItem({
  authorId,
  authorName,
  role,
  company,
  paragraphs,
}: RecommendationsType) {
  return (
    <article className={`my-6 sm:my-10`}>
      <header className="grid grid-cols-[auto_1fr] gap-x-3">
        <Image
          className="rounded-full"
          src={`/recommendations/${authorId}.jpeg`}
          width={64}
          height={64}
          alt={`Photo of the author (${authorName})`}
        />
        <div>
          <h2 className="text-base sm:text-lg md:text-xl xl:text-2xl font-bold">
            {authorName}
          </h2>
          <h3 className="text-sm sm:text-base md:text-lg xl:text-xl">
            {role} @ {company}
          </h3>
        </div>
      </header>
      <section className="mt-4 pl-4 border-l">
        {paragraphs.map((paragraph, i) => (
          <blockquote
            className="mb-4 text-sm xl:text-base text-justify"
            key={i}
          >
            {paragraph}
          </blockquote>
        ))}
      </section>
    </article>
  );
}
