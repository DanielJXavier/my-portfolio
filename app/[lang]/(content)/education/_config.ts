export const education = [
  {
    schoolId: "estacio",
    schoolName: "Est√°cio",
    years: "2020 ~ 2021",
  },
  {
    schoolId: "fumec",
    schoolName: "Universidade FUMEC",
    years: "2010 ~ 2014",
  },
] as const;
