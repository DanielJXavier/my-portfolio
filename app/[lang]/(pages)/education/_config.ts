export const education = [
  {
    schoolId: "estacio",
    schoolName: "Est√°cio",
    startYear: "2020",
    endYear: "2021",
  },
  {
    schoolId: "fumec",
    schoolName: "Universidade FUMEC",
    startYear: "2010",
    endYear: "2014",
  },
] as const;
