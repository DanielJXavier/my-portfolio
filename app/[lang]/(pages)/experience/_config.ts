export interface ExperienceInterface {
  key: string;
  companyId: string;
  companyName: string;
  startDate: string;
  endDate: string | null;
  mainStack: string[];
  resume: boolean;
  resumeAccomplishments: number[];
  hasBlackLogo: boolean;
}

export const experience: ExperienceInterface[] = [
  {
    key: "danielxavier-1",
    companyId: "danielxavier",
    companyName: "<DanielXavier />",
    startDate: "05-2024",
    endDate: null,
    mainStack: [
      "react",
      "next",
      "tailwind",
      "node",
      "ts",
      "jest",
      "cypress",
      "html",
      "css",
      "js",
      "contentful",
      "docker",
      "vercel",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: true,
  },
  {
    key: "pdsacademy-1",
    companyId: "pdsacademy",
    companyName: "PDSAcademy",
    startDate: "06-2025",
    endDate: "06-2025",
    mainStack: [
      "react",
      "next",
      "tailwind",
      "html",
      "css",
      "js",
      "functional-programming",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: true,
  },
  {
    key: "olx-4",
    companyId: "olx",
    companyName: "OLX",
    startDate: "01-2024",
    endDate: "04-2024",
    mainStack: [
      "react",
      "node",
      "kong",
      "s3",
      "firebase",
      "android",
      "kotlin",
      "ios",
      "swift-ui",
      "rn",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 4],
    hasBlackLogo: false,
  },
  {
    key: "olx-3",
    companyId: "olx",
    companyName: "OLX",
    startDate: "07-2023",
    endDate: "12-2023",
    mainStack: [
      "python",
      "es",
      "react",
      "storybook",
      "s3",
      "androind",
      "kotlin",
      "ios",
      "swift-ui",
      "rn",
      "postgre",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 4, 5],
    hasBlackLogo: false,
  },
  {
    key: "olx-2",
    companyId: "olx",
    companyName: "OLX",
    startDate: "09-2022",
    endDate: "06-2023",
    mainStack: [
      "react",
      "storybook",
      "s3",
      "android",
      "kotlin",
      "ios",
      "swift-ui",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 3],
    hasBlackLogo: false,
  },
  {
    key: "olx-1",
    companyId: "olx",
    companyName: "OLX",
    startDate: "08-2021",
    endDate: "08-2022",
    mainStack: [
      "html",
      "css",
      "js",
      "s3",
      "firebase",
      "android",
      "java",
      "kotlin",
      "ios",
      "swift-ui",
      "ad-manager",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "meaning-1",
    companyId: "meaning",
    companyName: "Meaning",
    startDate: "03-2022",
    endDate: "05-2023",
    mainStack: ["html", "css", "js", "bpmn", "gcp"],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "softplan-3",
    companyId: "softplan",
    companyName: "Softplan",
    startDate: "10-2020",
    endDate: "07-2021",
    mainStack: [
      "react",
      "micro-frontend",
      "dotnet",
      "graphql",
      "delphi",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "softplan-2",
    companyId: "softplan",
    companyName: "Softplan",
    startDate: "09-2019",
    endDate: "10-2020",
    mainStack: [
      "react",
      "micro-frontend",
      "dotnet",
      "graphql",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2, 4],
    hasBlackLogo: false,
  },
  {
    key: "softplan-1",
    companyId: "softplan",
    companyName: "Softplan",
    startDate: "12-2018",
    endDate: "08-2019",
    mainStack: [
      "react",
      "micro-frontend",
      "dotnet",
      "graphql",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "hotmart-3",
    companyId: "hotmart",
    companyName: "Hotmart",
    startDate: "06-2018",
    endDate: "11-2018",
    mainStack: ["vue", "nuxt", "node", "mysql", "html", "css", "js", "aws"],
    resume: false,
    resumeAccomplishments: [1, 2],
    hasBlackLogo: false,
  },
  {
    key: "hotmart-2",
    companyId: "hotmart",
    companyName: "Hotmart",
    startDate: "06-2017",
    endDate: "05-2018",
    mainStack: [
      "vue",
      "nuxt",
      "node",
      "mysql",
      "html",
      "css",
      "js",
      "php",
      "wordpress",
      "nginx",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2],
    hasBlackLogo: false,
  },
  {
    key: "hotmart-1",
    companyId: "hotmart",
    companyName: "Hotmart",
    startDate: "12-2016",
    endDate: "05-2017",
    mainStack: [
      "html",
      "css",
      "js",
      "php",
      "wordpress",
      "nginx",
      "mysql",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [],
    hasBlackLogo: false,
  },
  {
    key: "sydle-2",
    companyId: "sydle",
    companyName: "Sydle",
    startDate: "04-2014",
    endDate: "11-2016",
    mainStack: [
      "node",
      "express",
      "loopback",
      "angular",
      "html",
      "css",
      "js",
      "php",
      "wordpress",
      "magento",
      "apache",
      "bpmn",
      "linux",
      "windows",
      "oracle",
      "sql-server",
      "mysql",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [],
    hasBlackLogo: true,
  },
  {
    key: "sydle-1",
    companyId: "sydle",
    companyName: "Sydle",
    startDate: "04-2012",
    endDate: "03-2014",
    mainStack: [
      "js",
      "bpmn",
      "linux",
      "windows",
      "oracle",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [],
    hasBlackLogo: true,
  },
];
