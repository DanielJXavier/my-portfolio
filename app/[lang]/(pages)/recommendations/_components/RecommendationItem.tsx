import Image from "next/image";

type RecommendationItemPropsType = Readonly<{
  authorId: string;
  authorName: string;
  role: string;
  company: string;
  paragraphs: string[];
  imageAltText: string;
}>;

export default function RecommendationItem({
  authorId,
  authorName,
  role,
  company,
  paragraphs,
  imageAltText,
}: RecommendationItemPropsType) {
  return (
    <article className={`my-6 sm:my-10`} data-testid="recommendation-item">
      <header className="grid grid-cols-[auto_1fr] gap-x-3">
        <Image
          className="rounded-full"
          src={`/images/recommendations/${authorId}.jpeg`}
          width={64}
          height={64}
          alt={`${imageAltText} (${authorName})`}
        />

        <div>
          <h2 className="text-base sm:text-lg md:text-xl xl:text-2xl font-bold">
            {authorName}
          </h2>

          <h3 className="text-sm sm:text-base md:text-lg xl:text-xl">
            {role} @ {company}
          </h3>
        </div>
      </header>

      <section className="mt-4 pl-4 border-l">
        {paragraphs.map((paragraph, i) => (
          <blockquote
            className="mb-4 text-sm xl:text-base text-justify"
            key={`paragraph-${i}`}
          >
            {paragraph}
          </blockquote>
        ))}
      </section>
    </article>
  );
}
