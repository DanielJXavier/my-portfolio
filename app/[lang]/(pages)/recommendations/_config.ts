export const recommendations = [
  {
    authorId: "1758032899095",
    authorName: "Ben Lechlitner",
    company: "Lovevery",
  },
  {
    authorId: "1679942199683",
    authorName: "Nayane Menezes",
    company: "AB InBev",
  },
  {
    authorId: "1719021291674",
    authorName: "Wallace Freitas",
    company: "NEXT SI",
  },
  {
    authorId: "1716417055984",
    authorName: "Renan Silva",
    company: "Acert Grup",
  },
  {
    authorId: "1665746687624",
    authorName: "Camilla Polli",
    company: "OLX",
  },
  {
    authorId: "1711736735188",
    authorName: "Fernanda Rabelo",
    company: "OLX",
  },
  {
    authorId: "1692061488021",
    authorName: "Matheus Queiroz",
    company: "OLX",
  },
  {
    authorId: "1695594745530",
    authorName: "Douglas Ferreira",
    company: "OLX",
  },
  {
    authorId: "1565735422940",
    authorName: "Alan Cezar",
    company: "OLX",
  },
  {
    authorId: "1593552193665",
    authorName: "Matheus Missaci",
    company: "OLX",
  },
  {
    authorId: "1660079652443",
    authorName: "Ricardo Moraes",
    company: "VendorSmart",
  },
  {
    authorId: "1622852709278",
    authorName: "Mateus Queiroz Correia",
    company: "Nubank",
  },
  {
    authorId: "1606086729239",
    authorName: "Fernanda Ribeiro Durbano",
    company: "OLX",
  },
  {
    authorId: "1683719518610",
    authorName: "Gustavo Chimenes Dias",
    company: "OLX",
  },
  {
    authorId: "1712709978402",
    authorName: "CÃ¡ssio Fonseca",
    company: "Linx",
  },
  {
    authorId: "1695221396520",
    authorName: "Pedro Dias",
    company: "OLX",
  },
  {
    authorId: "1700516452017",
    authorName: "Euler Filho",
    company: "Techifide Ltd",
  },
  {
    authorId: "1713311445833",
    authorName: "Bruno Cechet",
    company: "Altruist",
  },
] as const;
