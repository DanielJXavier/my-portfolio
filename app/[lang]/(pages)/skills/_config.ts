export interface SkillInterface {
  key: string;
  name: string;
  size: 1 | 2 | 3 | 4 | 5;
}

export const hardSkills: SkillInterface[] = [
  /* Front-End */
  { key: "html", name: "HTML5", size: 5 },
  { key: "css", name: "CSS3", size: 5 },
  { key: "js", name: "JavaScript", size: 5 },
  { key: "react", name: "React", size: 4 },
  { key: "next", name: "Next.js", size: 3 },
  { key: "tailwind", name: "Tailwind CSS", size: 4 },
  { key: "vue", name: "Vue.js", size: 1 },
  { key: "nuxt", name: "Nuxt.js", size: 1 },
  { key: "angular", name: "Angular", size: 1 },
  { key: "responsive-design", name: "Responsive Design", size: 4 },
  { key: "seo", name: "SEO", size: 3 },
  { key: "ssr", name: "SSR", size: 2 },
  { key: "sass", name: "Sass", size: 4 },
  { key: "semantical-html", name: "Semantical HTML", size: 4 },
  { key: "ts", name: "TypeScript", size: 4 },
  { key: "webpack", name: "Webpack", size: 1 },
  { key: "redux", name: "Redux", size: 1 },
  { key: "jquery", name: "jQuery", size: 1 },
  { key: "acessibility", name: "Web Accessibility", size: 3 },
  { key: "web-vitals", name: "Core Web Vitals", size: 4 },
  { key: "performance", name: "Web Performance", size: 2 },
  { key: "security", name: "Web Security", size: 2 },
  { key: "web-sockets", name: "Web Sockets", size: 1 },
  { key: "styled-components", name: "Styled Components", size: 1 },
  { key: "pwa", name: "PWA", size: 2 },
  /* Back-End */
  { key: "node", name: "Node.js", size: 3 },
  { key: "express", name: "Express", size: 3 },
  { key: "loopback", name: "LoopBack", size: 1 },
  { key: "meteor", name: "Meteor", size: 1 },
  /* Test  */
  { key: "jest", name: "Jest", size: 3 },
  { key: "coverage", name: "Code Coverage", size: 3 },
  { key: "react-testing-library", name: "React Testing Library", size: 3 },
  { key: "cypress", name: "Cypress", size: 2 },
  { key: "chromatic", name: "Chromatic", size: 1 },
  /* APIs */
  { key: "http", name: "HTTP", size: 3 },
  { key: "graphql", name: "GraphQL", size: 1 },
  { key: "rest", name: "REST", size: 1 },
  /* PHP */
  { key: "php", name: "PHP", size: 1 },
  { key: "wordpress", name: "WordPress", size: 1 },
  { key: "magento", name: "Magento", size: 1 },
  { key: "laravel", name: "Laravel", size: 1 },
  /* Web Servers */
  { key: "nginx", name: "Nginx", size: 3 },
  { key: "apache", name: "Apache", size: 2 },
  /* CI / CD */
  { key: "github-actions", name: "GitHub Actions", size: 2 },
  { key: "bash", name: "Bash", size: 1 },
  { key: "docker", name: "Docker", size: 1 },
  { key: "yml", name: "YML", size: 1 },
  { key: "aws", name: "AWS", size: 2 },
  { key: "k8s", name: "Kubernets", size: 1 },
  { key: "terraform", name: "Terraform", size: 1 },
  /* Database */
  { key: "sql", name: "SQL", size: 3 },
  { key: "mysql", name: "MySQL", size: 1 },
  { key: "sql-server", name: "SQL Server", size: 1 },
  { key: "postgre", name: "PostgreSQL", size: 1 },
  { key: "oracle", name: "Oracle", size: 1 },
  { key: "nosql", name: "NoSQL", size: 2 },
  { key: "mongo", name: "MongoDB", size: 1 },
  /* Management Tools */
  { key: "jira", name: "Jira", size: 1 },
  { key: "confluence", name: "Confluence", size: 1 },
  { key: "slack-workflos", name: "Slack Workflows", size: 1 },
  { key: "slack-apps", name: "Slack Apps", size: 1 },
  { key: "bpmn", name: "BPMN", size: 1 },
  /* Business */
  { key: "a_b-testing", name: "A/B Testing", size: 1 },
  { key: "atomic-design", name: "Atomic Design", size: 1 },
  { key: "design-systems", name: "Design Systems", size: 1 },
  { key: "semver", name: "Semantic Versioning", size: 1 },
  { key: "ui_ux", name: "UI/UX", size: 1 },
  /* Other */
  { key: "functional-programming", name: "Functional Programming", size: 1 },
  { key: "oop", name: "OOP", size: 1 },
  { key: "regex", name: "Regular Expressions", size: 2 },
  { key: "markdown", name: "MarkDown", size: 1 },
];

export const softSkills = [
  { key: "communication", size: 5 },
  { key: "leadership", size: 5 },
  { key: "problem-solving", size: 5 },
  { key: "decision-making", size: 5 },
  { key: "empathy", size: 5 },
  { key: "conflict-resolution", size: 4 },
  { key: "adaptability", size: 4 },
  { key: "time-management", size: 3 },
  { key: "organization", size: 3 },
  { key: "strategic-thinking", size: 3 },
  { key: "mentoring", size: 4 },
  { key: "negotiation", size: 4 },
  { key: "resilience", size: 5 },
  { key: "critical-thinking", size: 3 },
  { key: "creativity", size: 3 },
  { key: "active-listening", size: 3 },
  { key: "team-building", size: 2 },
  { key: "ethical-decision-making", size: 1 },
  { key: "customer-orientation", size: 3 },
  { key: "learning-agility", size: 3 },
  { key: "curiosity", size: 4 },
  { key: "patience", size: 1 },
  { key: "public-speaking", size: 1 },
  { key: "networking", size: 1 },
  { key: "multitasking", size: 1 },
  { key: "attention-to-detail", size: 1 },
  { key: "motivation", size: 1 },
  { key: "persuasion", size: 1 },
  { key: "emotional-intelligence", size: 1 },
] as const;
