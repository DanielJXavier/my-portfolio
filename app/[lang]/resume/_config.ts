export const maxSummaryParagraphs = 3;

export const chosenHardSkills = [
  "react",
  "ts",
  "next",
  "node",
  "js",
  "aws",
  "tailwind",
  "contentful",
  "docker",
];

export const chosenSoftSkills = [
  "leadership",
  "stakeholder-management",
  "problem-solving",
];
