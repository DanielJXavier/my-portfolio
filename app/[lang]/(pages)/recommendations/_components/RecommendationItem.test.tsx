import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import RecommendationItem from "./RecommendationItem";

import { recommendations } from "../_config";
import { recommendations as recommendationsStrings } from "dictionaries/en.json";

describe("RecommendationItem component", () => {
  it("renders the component", () => {
    const { authorId, authorName, company } = recommendations[0];

    const { container } = render(
      <RecommendationItem
        authorId={authorId}
        authorName={authorName}
        role={recommendationsStrings.items[authorId].role}
        company={company}
        paragraphs={recommendationsStrings.items[authorId].paragraphs}
        imageAltText={recommendationsStrings.imageAltText}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
