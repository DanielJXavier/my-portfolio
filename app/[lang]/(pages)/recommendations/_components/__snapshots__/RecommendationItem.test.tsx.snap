// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationItem component renders the component 1`] = `
<div>
  <article
    class="my-6 sm:my-10"
    data-testid="recommendation-item"
  >
    <header
      class="grid grid-cols-[auto_1fr] gap-x-3"
    >
      <img
        alt="Photo of the author (Renan Silva)"
        class="rounded-full"
        data-nimg="1"
        decoding="async"
        height="64"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Frecommendations%2F1716417055984.jpeg&w=128&q=75"
        srcset="/_next/image?url=%2Fimages%2Frecommendations%2F1716417055984.jpeg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Frecommendations%2F1716417055984.jpeg&w=128&q=75 2x"
        style="color: transparent;"
        width="64"
      />
      <div>
        <h2
          class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold"
        >
          Renan Silva
        </h2>
        <h3
          class="text-sm sm:text-base md:text-lg xl:text-xl"
        >
          Software Engineer
           @ 
          Acert Grup
        </h3>
      </div>
    </header>
    <section
      class="mt-4 pl-4 border-l"
    >
      <blockquote
        class="mb-4 text-sm xl:text-base text-justify"
      >
        I had the pleasure of working with Daniel Xavier at MEANING, and I can confidently say that he is an exceptional professional. Daniel stood out as a senior full-stack developer focused on front-end, with expertise in technologies such as React, Next.js, TypeScript, Node.js, and JavaScript. His technical ability is always far ahead of other team members.
      </blockquote>
      <blockquote
        class="mb-4 text-sm xl:text-base text-justify"
      >
        In addition to his advanced technical knowledge, Daniel is a proactive collaborator, always presenting innovative ideas that add value to projects. He has a natural ability to lead and inspire the team, creating a collaborative and productive work environment.
      </blockquote>
      <blockquote
        class="mb-4 text-sm xl:text-base text-justify"
      >
        I highly recommend Daniel Xavier for any position he applies for. I have full confidence that he will be a valuable addition to any team, bringing with him not only exceptional technical skills but also a positive attitude and exemplary work ethic.
      </blockquote>
    </section>
  </article>
</div>
`;
